src/test/java/com/connexta/store/controllers/StoreControllerIngestTests.java

```diff
@@ -0,0 +1,334 @@
+
+    verifyNoInteractions(mockStoreService);
+  }
+
+  /**
+   * TODO Test RuntimeException and Throwable thrown by {@link StoreService#ingest(Long, String,
+   * InputStream, String, Long, InputStream)}
+   */
+  @ParameterizedTest(name = "{0} is the response status code when IngestService#ingest throws {1}")
+  @MethodSource("exceptionThrownByIngestServiceAndExpectedResponseStatus")
+  public void testIngestServiceExceptions(
+      final Throwable throwable, final HttpStatus expectedResponseStatus) throws Exception {
+    doThrow(throwable).when(mockStoreService).ingest(any(), any(), any(), any(), any(), any());
+
+    mockMvc
+        .perform(
+            multipart("/ingest")
+                .file(FILE)
+                .file(METACARD)
+                .param("correlationId", CORRELATION_ID)
+                .header(LAST_MODIFIED, LAST_MODIFIED_DATE)
+                .header("Accept-Version", ACCEPT_VERSION))
+        .andExpect(status().is(expectedResponseStatus.value()));
+  }
+
+  private static Stream<Arguments> exceptionThrownByIngestServiceAndExpectedResponseStatus() {
+    return Stream.of(
+        Arguments.of(
+            new StoreException("Test", new Throwable()), HttpStatus.INTERNAL_SERVER_ERROR));
+  }
+
+  @NotNull
+  private static InputStream inputStreamContentEq(@NotNull final InputStream expected) {
+    return argThat(
+        actual -> {
+          try {
+            return IOUtils.contentEquals(expected, actual);
+          } catch (final IOException e) {
+            fail("Unable to compare input streams", e);
+            return false;
+          }
+        });
+  }
+
+  @ParameterizedTest
+  @NullAndEmptySource
+  @ValueSource(strings = " ")
+  public void testIngestBlankContentType(final String contentType) {
+    when(mockFile.getContentType()).thenReturn(contentType);
+    assertThat(
+        storeController
+            .ingest(
+                ACCEPT_VERSION,
+                OffsetDateTime.parse(LAST_MODIFIED_DATE),
+                mockFile,
+                CORRELATION_ID,
+                METACARD)
+            .getStatusCode(),
+        is(HttpStatus.ACCEPTED));
+  }
+
+  @ParameterizedTest
+  @NullAndEmptySource
+  @ValueSource(strings = "   ")
+  public void testBadFilename(final String filename) {
+    when(mockFile.getContentType()).thenReturn("testContentType");
+    when(mockFile.getOriginalFilename()).thenReturn(filename);
+    assertThat(
+        storeController
+            .ingest(
+                ACCEPT_VERSION,
+                OffsetDateTime.parse(LAST_MODIFIED_DATE),
+                mockFile,
+                CORRELATION_ID,
+                METACARD)
+            .getStatusCode(),
+        is(HttpStatus.ACCEPTED));
+  }
+
+  @Test
+  void testIngestFileTooLarge() {
+    when(mockFile.getContentType()).thenReturn("testContentType");
+    when(mockFile.getOriginalFilename()).thenReturn("testFilename");
+    when(mockFile.getSize()).thenReturn(MAX_FILE_BYTES + 1);
```

Comment:
I don't think think this tests what will actually happen. The file size validation will happen on the Spring side before our controller.